plugins/modules/vm_portgroup_info.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Ansible Cloud Team (@ansible-collections)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: vm_portgroup_info
+version_added: '1.4.0'
+short_description: Returns information about the portgroups of virtual machines
+description:
+    - Returns information about the standard or distributed portgroups of virtual machines.
+author:
+    - Ansible Cloud Team (@ansible-collections)
+requirements:
+    - vSphere Automation SDK
+options:
+    vm_names:
+        description:
+            - VM names for retrieving the information about portgroup
+        required: true
+        type: list
+        elements: str
+extends_documentation_fragment:
+    - vmware.vmware.vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather list of portgroup by VMs
+  vmware.vmware.portgroup_info:
+    hostname: "https://vcenter"
+    username: "username"
+    password: "password"
+    vm_names:
+      - vm-test1
+      - vm-test2
+'''
+
+RETURN = r'''
+vm_portgroup_info:
+    description:
+        - Dictionary of the requested VMs with the portgroup information
+    returned: On success
+    type: dict
+    sample: {
+        "vm1": [
```

Comment:
The standard porgroup can be filtered by cluster or ESXi but not sure how to involve it here regarding the distributed is not related to a specific of ESXi but a group of ESXis.... 
Do you have any idea how can I do it?

Reply:
In that case maybe just having a cluster as a parameter to filter instead of just `vm_names` should be enough I guess. So seems the interface is OK, for future extensions.